dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/PasswordNullifier.java

```diff
@@ -59,6 +72,36 @@ public static String nullify(String value, byte delimiter) {
         return value;
     }
 
+    /**
+     * Scan the input value for the "password=" and "client_secret=" key markers
+     * and convert the password value to a series of *s. The delimiter value
+     * can be used if the search string is a sequence like
+     * "key=value<delim>key2=value2".
+     *
+     * @param value
+     * @param delimiter
+     * @return String
+     */
+    public static String nullify(String header, String value, byte delimiter) {
+        // check to see if we need to null out passwords
+        if (null == value) {
+            return null;
+        }
+
+        String source = value.toLowerCase();
+        StringBuilder b = new StringBuilder(value);
+        boolean modified = optionallyMaskChars(b, source, delimiter, PASSWORD_PATTERN);
+        modified = optionallyMaskChars(b, source, delimiter, CLIENT_SECRET_PATTERN) || modified;
+        if (header.equalsIgnoreCase(com.ibm.wsspi.http.channel.values.HttpHeaderKeys.HDR_AUTHORIZATION.getName())) {
```

Comment:
here?  I am clicking weird place.  Sorry @wtlucy  if many notifications were sent.   Hope this looks better.